parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -632,11 +645,30 @@ pub mod pallet {
 			Self::deposit_event(Event::BeaconHeaderImported{block_hash: block_root, slot: slot});
 		}
 
-		fn store_execution_header(block_root: H256, header: ExecutionHeaderOf<T>) {
+		fn store_execution_header(block_root: H256, header: ExecutionHeaderOf<T>, slot: u64, beacon_header_hash: H256) {
 			let block_number = header.block_number;
 
 			<ExecutionHeaders<T>>::insert(block_root, header);
 
+			let mut execution_header_state = <LatestExecutionHeaderState<T>>::get();
```

Comment:
I think the idiom in Substrate is to use [StorageValue::try_mutate](https://crates.parity.io/frame_support/storage/trait.StorageValue.html#tymethod.try_mutate) when a storage item optionally needs to be mutated.

Reply:
Example usage here: https://github.com/Snowfork/snowbridge/blob/3eddd6785eaca4402cb56d835886d98a0442d9e8/parachain/pallets/asset-registry/src/lib.rs#L54

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/696/commits/ebcc39e7a18f57a090c79fad496d192a96920031.

Reply:
Oops, looks I was wrong. 🤦

try_mutate is for when an error could be returned during between the read and write. In your case, no error could ever be returned. So your previous code was actually fine. Sorry.

Reply:
Maybe best to just revert https://github.com/Snowfork/snowbridge/commit/ebcc39e7a18f57a090c79fad496d192a96920031